text/0000-ember-language-tooling-roadmap.md

```diff
@@ -0,0 +1,169 @@
+
+### Proposal 4: Migrate Glint to provide editor completions, type-checking, etc, using a TS Plugin
+
+Vue Language Tooling has deprecated their Language-Server-backed language tooling in favor of a TS Plugin -driven tooling (while continuing to maintain vue-tsc for actually compilation of files).
+
+There are many benefits to shifting towards a TS Plugin approach:
+
+1. Other non-Ember/Glint-specific TS Plugins can be used in conjunction with Glint's future TS Plugin.
+  - i.e. by expressing .gts files as TS Plugin, we could use plugins/libraries like:
+    - esm.sh (https://twitter.com/jexia_/status/1733032441521308014)
+    - TSL (https://github.com/johnsoncodehk/typescript-linter)
+2. In general, TS Plugins allow custom file types to be recognizable to other custom file types that are supported by TS Plugins
+  - e.g a TS Plugin for importing xstate state machines, or .yml files, can work with .gts files, and if some other language wanted to import a `.gts` file, Glint's `.gts` file TS Plugin would facilitate that in a manner that the LSP approach would not.
+3. Somewhat anecdotally, TS Plugins seem to offer better performance than language servers even though they're internally performing the same work
+
+### Proposal 5: Merge Glint and Ember Language Server together into a singular "Ember Language Tooling" package
+
+This is another major milestone and perhaps worth another RFC, but as it stands, notwithstanding the effort involved to merge one into the other, there is no benefit and only confusion in Ember developers having to install something called "Ember Language Server" AND "Glint" to provide them modern Ember.js language tooling, so ideally these should be merged into one.
+
+I will leave this part underspecified for now -- it'll be easier to elaborate on what needs to happen here after we've completed our work to re-work Glint atop of Volar.
+
+### Proposal 6: Move Glint to the "ember-tooling" Github Organization.
+
+Regardless of how Proposal 5 is resolved, we should consolidate Ember tooling into the "ember-tooling" organization, rather than having things sprawled between "ember-tooling" and "typed-ember".
+
+### Proposal 7: Strongly encourage the deprecation and removal of sprawling editor extensions with overlapping functionality in the ecosystem
+
+With a unified Glint + ELS, we encourage maintainers of extensions providing Ember syntaxes / tooling to deprecate and ultimately remove their extensions from the VSCode marketplace and archive their Github repos, directing their traffic to the Ember Language Tooling package.
+
+## How we teach this
+
+(This RFC section not applicable.)
+
+## Drawbacks
+
+There is some risk to placing such a heavy bet on Volar in case they take things in a direction that is not advisable or incompatible/inconvenient to Ember. If Volar takes things in an unseemly direction, we would have to use and maintain a fork of Volar.
```

Comment:
I don't see this as a big risk, and even if this happens we can fork an Emlar or Vober version and we'll be back a the status quo of providing our own ecosystem specific tooling.